plugins/scaffolder-backend-module-gitlab/src/actions/gitlabMergeRequest.ts

```diff
@@ -238,12 +277,26 @@ export const createPublishGitlabMergeRequestAction = (options: {
         }
       }
 
-      const actions: Types.CommitAction[] = fileContents.map(file => ({
-        action: getFileAction(
-          { file, targetPath },
-          remoteFiles,
-          ctx.input.commitAction,
-        ),
+      const actions: Types.CommitAction[] = (
+        (
+          await Promise.all(
+            fileContents.map(async file =>
+              getFileAction(
+                { file, targetPath },
+                { repoID, branch: targetBranch! },
+                api,
+                ctx,
+                remoteFiles,
```

Comment:
If the input.commitAction === 'skip', should we even bother trying to fetch the remote files to save us the extra calls? Would it make sense to skip this at line 266 where if input.commitAction !== 'skip' we try to fetch?

Reply:
Technically that should already be accounted for at the line you cite since that tests for `auto` commit action behavior, but it does make sense to skip the action check entirely for `skip`.